lib/health_check/search_checker.rb

```diff
@@ -1,14 +1,16 @@
 require "uri"
 require "health_check/check_file_parser"
 require "health_check/calculator"
+require "health_check/search_check_report"
 
 module HealthCheck
   class SearchChecker
     attr_reader :search_client
 
-    def initialize(options = {})
-      @test_data_file = options[:test_data]
-      @search_client = options[:search_client]
+    def initialize(search_client:, test_data:, produce_report: true)
+      @test_data_file = test_data
+      @search_client  = search_client
+      @file_output    = produce_report ? SearchCheckReport.new : File.open(File::NULL, 'w')
```

Comment:
Instead of writing to null, can we not write at all when the report hasn't been requested.